### DISCUSSION TITLE
Not sure if im just missing something, but a question on the "onSortingChange" prop

### DISCUSSION THREAD

**Discussion by Incin** [KevinVandy/material-react-table#452] [Q&A] [answered]
I have a custom function that I am running on the onSortingChange prop and I noticed when you do the "clear sorting" from the column toolbar, it returns just an empty array. Is there no way to get the id of the column that is being cleared so I can remove it from my state?

Thanks so much for the help!

**Comment by KevinVandy** [comment, marked as answer]
There seems to be a lot of confusion for how all of the on[State]Change props work. They are only for hoisting state up for you to manage and read. It is explained in depth in the state management docs
https://www.material-react-table.com/docs/guides/table-state-management

**Reply by Incin** [reply]
Yeah, I actually figured this out and forgot to update it here. Thanks for the post!
